op-node/rollup/derive/calldata_source.go

```diff
@@ -76,10 +75,11 @@ func (ds *CalldataSource) Next(ctx context.Context) (eth.Data, error) {
 // DataFromEVMTransactions filters all of the transactions and returns the calldata from transactions
 // that are sent to the batch inbox address from the batch sender address.
 // This will return an empty array if no valid transactions are found.
-func DataFromEVMTransactions(dsCfg DataSourceConfig, batcherAddr common.Address, txs types.Transactions, log log.Logger) []eth.Data {
+func DataFromEVMTransactions(dsCfg DataSourceConfig, batcherAddr common.Address, gTxs []eth.GenericTx, log log.Logger) []eth.Data {
 	out := []eth.Data{}
-	for _, tx := range txs {
-		if isValidBatchTx(tx, dsCfg.l1Signer, dsCfg.batchInboxAddress, batcherAddr, log) {
+	for _, gTx := range gTxs {
+		if isValid := isValidBatchTx(gTx, dsCfg.l1Signer, dsCfg.batchInboxAddress, batcherAddr, log); isValid {
+			tx, _ := gTx.Transaction() // ignoring error because isValid implies it is a valid transaction
```

Comment:
It could still happen that a transaction looks like a valid one at first (because of a first byte in the accepted range), but then decoding fails. Node that only the `Transaction()` call decodes. So I think what we should do here, to avoid panics during derivation, is to log a warning if we get an error here, then drop this tx and go on to the next in the loop.

Reply:
Ah I see, thanks for clarifying.

Reply:
We could also let `isValidBatchTx` return `[]byte, bool` and directly return the data (and then rename the function). Then we'd only call it once.